presto-test-jdbc-compatibility/src/test/java/io/prestosql/TestJdbcCompatibility.java

```diff
@@ -0,0 +1,509 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logging;
+import io.prestosql.plugin.mongodb.MongoPlugin;
+import io.prestosql.server.testing.TestingPrestoServer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.Array;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Strings.repeat;
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.JdbcDriverCapabilities.driverVersion;
+import static io.prestosql.JdbcDriverCapabilities.hasBrokenParametricTimestampWithTimeZoneSupport;
+import static io.prestosql.JdbcDriverCapabilities.supportsParametricTimestamp;
+import static io.prestosql.JdbcDriverCapabilities.supportsParametricTimestampWithTimeZone;
+import static io.prestosql.JdbcDriverCapabilities.supportsSessionPropertiesViaConnectionUri;
+import static io.prestosql.JdbcDriverCapabilities.testedVersion;
+import static java.lang.Integer.min;
+import static java.lang.String.format;
+import static java.sql.Types.JAVA_OBJECT;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestJdbcCompatibility
```

Comment:
TestJdbcResultSetCompatibility seems to cover more, so it seems TestJdbcCompatibility
is redundant (but it isn't!).

Add javadoc explaining how this class is different from TestJdbcResultSetCompatibility